tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/jwt_test.yml

```
@@ -0,0 +1,54 @@
+- vars:
+    role_id: test-role
+    jwt: '{{ lookup("file", "token.jwt") }}'
+    jwt_invalid: '{{ lookup("file", "token_invalid.jwt") }}'
+  block:
+    - name: 'Fetch secrets using "hashi_vault" lookup'
+      set_fact:
+        secret1: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret1 auth_method=jwt jwt=' ~ jwt ~ ' role_id=' ~ role_id) }}"
+        secret2: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret2 auth_method=jwt jwt=' ~ jwt ~ ' role_id=' ~ role_id) }}"
+
+    - name: 'Fetch secret using "hashi_vault" lookup with VAULT_JWT environment variable'
+      set_fact:
+        secret4: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret4?version=2 auth_method=jwt role_id=' ~ role_id) }}"
+      args:
+        apply:
+          environment:
+            VAULT_JWT: '{{ jwt }}'
```

```suggestion
    - name: 'Fetch secret using "hashi_vault" lookup with VAULT_JWT environment variable'
      environment:
        VAULT_JWT: '{{ jwt }}'
      set_fact:
        secret4: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret4?version=2 auth_method=jwt role_id=' ~ role_id) }}"
```

I think the reason this test is failing is that `args: apply:` is specific to `include_task`. You should be able to just directly use the `environment:` keyword on this task.

---

I have tried this already, but I will give it another shot...